nucliadb_core/src/fs_state.rs

```diff
@@ -27,8 +27,17 @@ use std::time::SystemTime;
 use fs2::FileExt;
 use serde::de::DeserializeOwned;
 use serde::Serialize;
+use thiserror::Error;
 
-use super::DiskR;
+pub type FsResult<O> = std::result::Result<O, FsError>;
+
+#[derive(Debug, Error)]
+pub enum FsError {
```

Comment:
You have a point there, I  decided to wrapped it in case the module grows in the future, since going from `io::Error` to our own type error could require a lot of refactoring. Also I think wrapping it on our own type gives us the advantage of knowing when the error comes from our library, and thanks to `thiserror` we loose 0 in ergonomics.

Reply:
Kinda agree. I just feel like the scope of 'FsError' and 'DiskError' is not clear (particularly when 'io::Error' as the same responsability). 
But it's definitely not a blocker :cat: